bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -19,112 +22,188 @@ class fw(object):
+# Return an array of true or false if the trigger has not been
+# time-slid
+fore_locs = all_trigs.timeslide_id == 0
 logging.info("%s clustered foreground triggers" % fore_locs.sum())
 
 logging.info("Clustering coinc triggers (exclusive of zerolag)")
-e = e.cluster(args.cluster_window)
+exc_zero_trigs = exc_zero_trigs.cluster(args.cluster_window)
+
+# If a particular index of all_trigs.timeslide_id isn't 0, evaluate true.
+# So this returns an array of Booleans.
+back_locs = all_trigs.timeslide_id != 0
 
 logging.info("Dumping foreground triggers")
 f = fw(args.output_file)
-f.attrs['detector_1'] = d.attrs['detector_1']
-f.attrs['detector_2'] = d.attrs['detector_2']
-f.attrs['timeslide_interval'] = d.attrs['timeslide_interval']
+f.attrs['detector_1'] = all_trigs.attrs['detector_1']
+f.attrs['detector_2'] = all_trigs.attrs['detector_2']
+f.attrs['timeslide_interval'] = all_trigs.attrs['timeslide_interval']
 
+# Parse h_iterations to a string so we can use it later
+h_iter_string = str(h_iterations)
 # Copy over the segment for coincs and singles
-for key in d.seg.keys():
-    f['segments/%s/start' % key] = d.seg[key]['start'][:]
-    f['segments/%s/end' % key] = d.seg[key]['end'][:]
+for key in all_trigs.seg.keys():
+    f['segments_h/h_iter' + h_iter_string + '/%s/start' % key] = all_trigs.seg[key]['start'][:]
+    f['segments_h/h_iter' + h_iter_string + '/%s/end' % key] = all_trigs.seg[key]['end'][:]
 
 if fore_locs.sum() > 0:
-    f['segments/foreground_veto/start'] = veto_start
-    f['segments/foreground_veto/end'] = veto_end
-    for k in d.data:
-        f['foreground/' + k] = d.data[k][fore_locs]
+    f['segments_h/h_iter' + h_iter_string + '/foreground_veto/start'] = remove_start_time
+    f['segments_h/h_iter' + h_iter_string + '/foreground_veto/end'] = remove_end_time
+    for k in all_trigs.data:
+        f['foreground_h/h_iter' + h_iter_string + '/' + k] = all_trigs.data[k][fore_locs]
 else:
     # Put SOMETHING in here to avoid failures later
-    f['segments/foreground_veto/start'] = numpy.array([0])
-    f['segments/foreground_veto/end'] = numpy.array([0])
-    for k in d.data:
-        f['foreground/' + k] = numpy.array([], dtype=d.data[k].dtype)
-
-back_locs = d.timeslide_id != 0
+    f['segments_h/h_iter' + h_iter_string + '/foreground_veto/start'] = numpy.array([0])
+    f['segments_h/h_iter' + h_iter_string + '/foreground_veto/end'] = numpy.array([0])
+    for k in all_trigs.data:
+        f['foreground_h/h_iter/' + k] = numpy.array([], dtype=all_trigs.data[k].dtype)
 
+# If the sum of all Booleans in the array (1 = True, 0 = False) is False,
+# then there aren't any background triggers. So leave!
 if (back_locs.sum()) == 0:
     logging.warn("There were no background events, so we could not assign "
                  "any statistic values")
     sys.exit()
-    
+
 logging.info("Dumping background triggers (inclusive of zerolag)")
-for k in d.data:
-    f['background/' + k] = d.data[k][back_locs]
+for k in all_trigs.data:
```

Comment:
I haven't yet changed it back to 'background' though, but I can do that. I don't think it will interfere with any of the plots I'm trying to do on the wiki: https://www.lsc-group.phys.uwm.edu/ligovirgo/cbcnote/PyCBC/hierarchical-removal-dev-O2
since what I would call 'background_h0', would be equivalent to 'background' at the end of this code anyway. 

I'll edit this as soon as I can verify it doesn't mess up with the plotting I have downstream of the code.